src/schemas/ImmutableUtils.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * Helpers to enable Immutable compatibility *without* bringing in
+ * the 'immutable' package as a dependency.
+ */
+
+function stringifiedArray(array) {
+  return array.map((item) => item && item.toString());
+}
+
+/**
+ * Check if an object is immutable by checking if it implements the
+ * getIn method.
+ *
+ * @param  {any} object
+ * @return {bool}
+ */
+export function isImmutable(object) {
+  return !!(object && object.getIn);
```

Comment:
edge-case: what if someone has a key, `getIn` on their entity? (also true for `setIn` and `hasIn`.